### ISSUE TITLE
Add a directive to set the "coreutils" implementation

### ISSUE THREAD

**Issue by dgutson** [koalaman/shellcheck#2758]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

GNU Bash and GNU coreutils are two separate projects, which may or may not be installed together.

We should be able to override the coreutils implementation separate from the shell implementation,
for example

```sh
# shellcheck shell=bash utils=POSIX
```

or

```sh
# shellcheck shell=sh utils=GNU
```

For example, on Alpine Linux, you can install bash and keep busybox' sed implementation, meaning that despite you will get the bash shebang, the sed flags should be those of POSIX.
